code/modules/mob/living/simple_animal/hostile/terror_spiders/gray.dm

```diff
@@ -35,4 +41,102 @@
 			visible_message("<span class='danger'>[src] buries grey fangs deep into the [inject_target] of [target]!</span>")
 		else
 			visible_message("<span class='danger'>[src] bites [target], but cannot inject venom into their [inject_target]!</span>")
-		L.attack_animal(src)
\ No newline at end of file
+		L.attack_animal(src)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/gray/adjustBruteLoss(damage)
+	..(damage)
+	if(invisibility > 0 || icon_state == "terror_gray_cloaked")
```

Comment:
`icon_state ==`
This is gross. You should have a `cloaked` variable or something along those lines, or just not have the second part at all. Spiders shouldn't be able to end up in a state where their invisibility is unset but they are still using the cloaked sprite.